<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>Adapted and non-adapted HLA Mutations</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HLA_Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="050_AllHLAMutations.html">All HLA Mutations</a>
</li>
<li>
  <a href="051_AdaptedHLAMutations.html">Adapted HLA Mutations</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mnoguera/HLA_Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Adapted and non-adapted HLA Mutations</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">3/3/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HLA_Analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220303code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220303)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220303code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220303)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommnogueraHLAAnalysistree521de8e32107c8be83412fb3bdf481364095a3b1targetblank521de8ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mnoguera/HLA_Analysis/tree/521de8e32107c8be83412fb3bdf481364095a3b1" target="_blank">521de8e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommnogueraHLAAnalysistree521de8e32107c8be83412fb3bdf481364095a3b1targetblank521de8ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mnoguera/HLA_Analysis/tree/521de8e32107c8be83412fb3bdf481364095a3b1" target="_blank">521de8e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/RisVac_Sieve/
    Ignored:    data/HLA footprints for AELIX002 analyses without B15 imprints 3-2-2022.xls
    Ignored:    data/Mostres_Aelix_PreART_HLA_20200930_noID. with 2 digit HLA reduction 3-2-2022.xlsx
    Ignored:    data/Mostres_Aelix_PreART_HLA_20200930_noID. with 2 digit HLA reduction 3-2-2022_vMNJ_1.0.csv
    Ignored:    data/Mostres_Aelix_PreART_HLA_20200930_noID. with 2 digit HLA reduction 3-2-2022_vMNJ_1.0.xlsx
    Ignored:    data/Optimal list 2022 with coordinates for ALX 002 analyses 3-2-2022 .xls
    Ignored:    data/VariantData_PASeq/

Untracked files:
    Untracked:  analysis/060_EpitopeSequences.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/050_AllHLAMutations.Rmd</code>) and HTML (<code>docs/050_AllHLAMutations.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/521de8e32107c8be83412fb3bdf481364095a3b1/analysis/050_AllHLAMutations.Rmd" target="_blank">521de8e</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
workflowr::wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/2102c289b5620736495be345342b890ee3d989a6/analysis/050_AllHLAMutations.Rmd" target="_blank">2102c28</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
remove tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mnoguera/HLA_Analysis/2102c289b5620736495be345342b890ee3d989a6/docs/050_AllHLAMutations.html" target="_blank">2102c28</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
remove tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mnoguera/HLA_Analysis/f600d53c3a06f37e1be24473f9b1dba49472790f/docs/050_AllHLAMutations.html" target="_blank">f600d53</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
update htmls
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/c26cff7bab5e6b1b6260cd87b94988489c9b7191/analysis/050_AllHLAMutations.Rmd" target="_blank">c26cff7</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
polish details
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mnoguera/HLA_Analysis/17c174c92e9f92eda0f9c1f824c91ed3f9edd347/docs/050_AllHLAMutations.html" target="_blank">17c174c</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/f9d9128976ad5a94196a0f97a924853aeee37802/analysis/050_AllHLAMutations.Rmd" target="_blank">f9d9128</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
workflowr::wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mnoguera/HLA_Analysis/f9d9128976ad5a94196a0f97a924853aeee37802/docs/050_AllHLAMutations.html" target="_blank">f9d9128</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
workflowr::wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/b76c6f9ec944576cefba3c1eb573b59520a27cba/analysis/050_AllHLAMutations.Rmd" target="_blank">b76c6f9</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
init analysis (adapted and non-adapted)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mnoguera/HLA_Analysis/b76c6f9ec944576cefba3c1eb573b59520a27cba/docs/050_AllHLAMutations.html" target="_blank">b76c6f9</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
<td>
init analysis (adapted and non-adapted)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(rootdir = here::here())</code></pre>
<div id="hla-specific-all-mutations" class="section level1">
<h1>HLA Specific / All Mutations</h1>
<div id="import-data" class="section level2">
<h2>Import Data</h2>
<p>Need to import data from different sources:</p>
<ul>
<li>HLA-Specific escape mutation list. This excel sheet contains information on mutations that are specific to each of HLA (2-digit resolution)</li>
<li>HLA from patients: Each patient has a specifc HLA setup composed of HLA-A, HLA-B and HLA-C, each of them consisting of two alleles.</li>
<li>Patient-Specific Variant list at the amino acid level, as obtained from Illumina Sequencing / PASeq.</li>
</ul>
<div id="hla-specific-mutations" class="section level3">
<h3>HLA-Specific mutations</h3>
<p>This excel sheet contains information on mutations that are specific to each of HLA (2-digit resolution). Containt the following relevant fields:</p>
<ul>
<li>Protein: Short code for HIV protein.</li>
<li>HLA: 2-digit HLA of that specific mutation.</li>
<li>Codon / position_in_protein: Codon-based coordinate of that mutation within Protein.</li>
<li>AAHXB2/AminoAcid: Reference and variant amino acid.</li>
<li>EPITOPE_SEQUENCE: Is the mutation within a epitope sequence? (YES/NO)</li>
<li>IN.OUT.vaccine: Is the mutation within the vaccines immunogen?</li>
</ul>
<p>We di several things with this data:</p>
<ul>
<li>Generate a Mutation field in the shape of “GAG:T454R”</li>
<li>Specify HXB2 start and end coordinates, separately, this may be handy for cooridnate based analysis.</li>
<li>Generate a mutation_string new field, that contains all mutations for each HLA. This is not elegant but may be practical.</li>
</ul>
<pre class="r"><code>require(dplyr)</code></pre>
<pre><code>Loading required package: dplyr</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.1.2</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>require(gdata)</code></pre>
<pre><code>Loading required package: gdata</code></pre>
<pre><code>gdata: read.xls support for &#39;XLS&#39; (Excel 97-2004) files ENABLED.</code></pre>
<pre><code></code></pre>
<pre><code>gdata: read.xls support for &#39;XLSX&#39; (Excel 2007+) files ENABLED.</code></pre>
<pre><code>
Attaching package: &#39;gdata&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    nobs</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    object.size</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    startsWith</code></pre>
<pre class="r"><code>HLA_mutations&lt;-read.xls(&quot;data/HLA footprints for AELIX002 analyses without B15 imprints 3-2-2022.xls&quot;,sheet = 2)
HLA_mutations&lt;-HLA_mutations %&gt;% 
  dplyr::mutate(mutation=paste0(AAHXB2,position_in_protein,AminoAcid)) %&gt;% 
  dplyr::mutate(HXB2_start=strsplit(x = HXB2_location,split = &quot;..&quot;,fixed=T)[[1]][1]) %&gt;% 
  dplyr::mutate(HXB2_end=strsplit(x = HXB2_location,split = &quot;..&quot;,fixed=T)[[1]][2]) %&gt;% 
  dplyr::mutate(Mutation=paste0(Protein,&quot;:&quot;,AAHXB2,position_in_protein,AminoAcid)) %&gt;% 
  dplyr::group_by(HLA) %&gt;% 
  dplyr::mutate(mutation_string=paste0(Mutation,collapse=&quot;;&quot;)) 

# head(HLA_mutations) %&gt;% knitr::kable()</code></pre>
<pre class="r"><code>HLA_mutation_collapsed&lt;-HLA_mutations %&gt;% 
  select(HLA,mutation_string) %&gt;% 
  unique()

# head(HLA_mutations)</code></pre>
</div>
<div id="patients-hla" class="section level3">
<h3>Patients HLA</h3>
<p>We have an excel sheet that has been manually edited, containing 4D and 2D HLA information for each patient(SAmpleID). We basically add A/B/C to each allele to be able to merge downstream</p>
<pre class="r"><code>patient_hla&lt;-xlsx::read.xlsx(&quot;data/Mostres_Aelix_PreART_HLA_20200930_noID. with 2 digit HLA reduction 3-2-2022_vMNJ_1.0.xlsx&quot;,sheetIndex = 1) %&gt;% 
  dplyr::mutate(HLA.A1_2D=paste0(&quot;A&quot;,HLA.A1_2D)) %&gt;%
  dplyr::mutate(HLA.A2_2D=paste0(&quot;A&quot;,HLA.A2_2D)) %&gt;%
  dplyr::mutate(HLA.B1_2D=paste0(&quot;C&quot;,HLA.B1_2D)) %&gt;%
  dplyr::mutate(HLA.B2_2D=paste0(&quot;C&quot;,HLA.B2_2D)) %&gt;%
  dplyr::mutate(HLA.C1_2D=paste0(&quot;C&quot;,HLA.C1_2D)) %&gt;%
  dplyr::mutate(HLA.C2_2D=paste0(&quot;C&quot;,HLA.C2_2D))
  
# head(patient_hla)</code></pre>
</div>
<div id="patients-variants" class="section level3">
<h3>Patients Variants</h3>
<p>We obtain AA variants from PASeq analysis using self-consensus and codon-level variant calling only on pol, gag, vif and nef. We do several things:</p>
<ul>
<li>Keep those above 20% of the viral population.</li>
<li>Remove Deletion</li>
<li>All protein names to uppercase.</li>
<li>Add protein names to match the format above(“GAG:T454R”)</li>
<li>Change sample names to match format above.</li>
</ul>
<pre class="r"><code>require(dplyr)
temp = list.files(path = &quot;data/VariantData_PASeq/&quot;,pattern=&quot;*.csv&quot;,full.names = T)
variant_df&lt;-lapply(temp, read.csv) %&gt;% 
  data.table::rbindlist() %&gt;% 
  dplyr::filter(FREQ&gt;=20) %&gt;%
  tidyr::separate(FILE,into = c(&quot;SampleID&quot;,&quot;Garbage&quot;),sep = &quot;_S&quot;) %&gt;% 
  dplyr::select(-Garbage) %&gt;% 
  dplyr::mutate(PROTEIN=toupper(PROTEIN)) %&gt;% 
  dplyr::mutate(Mutation=paste0(PROTEIN,&quot;:&quot;,MUTATION)) %&gt;% 
  dplyr::filter(! grepl(&quot;Del&quot;,MUTATION,ignore.case = T)) %&gt;% 
  dplyr::mutate(SampleID=stringr::str_replace(SampleID,&quot;-&quot;,&quot;_&quot;))</code></pre>
<pre class="r"><code>metadata_df&lt;-xlsx::read.xlsx(&quot;../Pre-Treatment/Aelix-002_VG/data/raw/Dades_Aelix_x Seq_20210714.xlsx&quot;,sheetIndex = 1) %&gt;% 
  dplyr::rowwise() %&gt;% 
  dplyr::mutate(SampleID=paste0(&quot;Alx_&quot;,strsplit(ID1,&quot;\ &quot;,fixed=T)[[1]][2]))</code></pre>
</div>
</div>
<div id="data-merging" class="section level2">
<h2>Data Merging</h2>
<div id="hla-specifc-2d-all-mutations" class="section level3">
<h3>HLA-specifc | 2D | All mutations</h3>
<p>For each patient we select the mutations that are specific to his/her HLA and create a patient-specific mutation string that contains all mutations.</p>
<pre class="r"><code>require(dplyr)

for (patient in (patient_hla$SampleID)){
  aux_patient_hla_2D&lt;-(patient_hla %&gt;% 
    filter(SampleID==patient) %&gt;% 
    select(HLA.A1_2D,HLA.A2_2D,HLA.B1_2D,HLA.B2_2D,HLA.C1_2D,HLA.C2_2D) )[1,]
  patient_hla[patient_hla$SampleID==patient,&quot;mutation_string&quot;]&lt;-(HLA_mutations %&gt;% 
    select(-mutation_string) %&gt;% 
    dplyr::filter(HLA %in% aux_patient_hla_2D[1,]) %&gt;% 
    dplyr::group_by(.) %&gt;% 
    dplyr::mutate(mutation_string=paste0(Mutation,collapse=&quot;;&quot;)) %&gt;% 
    dplyr::mutate(SampleID=patient) %&gt;% 
    dplyr::select(SampleID,mutation_string) %&gt;% 
    unique())$mutation_string
}</code></pre>
<p>For each patient, we take all AA variants and take each one of them as being present in the patients mutations_string, thus indicating whether each mutation is specific of the patients HLA. Then we sum up the number of mutations in each patients that are found in their HLA and integrate patient metadata regarding Placebo/Vaccine group to nicely plot and run test the stats (Wilcoxon Rank sum).</p>
<p>Apparently there are no significant difference between groups regarding the number of HLA-specific mutations, including both ADAPTED and non-ADAPTED.</p>
<pre class="r"><code>require(ggplot2)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code># patient_hla$SampleID
# unique(variant_df$SampleID)

variant_df %&gt;% 
  left_join(patient_hla,by=&quot;SampleID&quot;) %&gt;% 
  dplyr::rowwise() %&gt;% 
  dplyr::mutate(withinHLA=ifelse(grepl(pattern=Mutation,x = mutation_string),1,0)) %&gt;% 
  group_by(SampleID) %&gt;% 
  dplyr::summarise(total_within_HLA=sum(withinHLA)) %&gt;% 
  left_join(metadata_df) %&gt;% 
  dplyr::rename(TreatmentGroup=Treatment.Vax.vs.Pbo,n_HLA_mutations=total_within_HLA) %&gt;% 
  ggplot(.,aes(x=TreatmentGroup,y=n_HLA_mutations,color=TreatmentGroup))+
    geom_boxplot()+
    ggpubr::stat_compare_means(method = &quot;wilcox.test&quot;,label.x = 1.3,label.y = 25)+
    theme_classic()+
    #geom_jitter(shape=16, position=position_jitter(0.2))+
    scale_color_manual(values=c(&quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
    ggtitle(&quot;(gag,pol,nef,vif) HLA-specific mutations&quot;,subtitle = &quot;Adapted and non-Adapted Mutations&quot;)+
    theme(
      plot.title = element_text(color=&quot;black&quot;, size=14, face=&quot;bold.italic&quot;),
      axis.title.x = element_text(color=&quot;darkgrey&quot;, size=12, face=&quot;bold&quot;),
      axis.title.y = element_text(color=&quot;darkgrey&quot;, size=12, face=&quot;bold&quot;)
    )+
    geom_dotplot(binaxis=&#39;y&#39;, stackdir=&#39;center&#39;, dotsize=0.8)</code></pre>
<pre><code>Joining, by = &quot;SampleID&quot;</code></pre>
<pre><code>Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/050_AllHLAMutations.Rmd/calculate%20numbe%20of%20variants%20in%20patients%20HLA%20Mutations%20List-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-calculate-numbe-of-variants-in-patients-HLA-Mutations-List-1">
Past versions of "calculate numbe of variants in patients HLA Mutations List-1.png"
</button>
</p>
<div id="fig-no-spaces-calculate-numbe-of-variants-in-patients-HLA-Mutations-List-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/f9d9128976ad5a94196a0f97a924853aeee37802/docs/figure/050_AllHLAMutations.Rmd/calculate numbe of variants in patients HLA Mutations List-1.png" target="_blank">f9d9128</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mnoguera/HLA_Analysis/blob/b76c6f9ec944576cefba3c1eb573b59520a27cba/docs/figure/050_AllHLAMutations.Rmd/calculate numbe of variants in patients HLA Mutations List-1.png" target="_blank">b76c6f9</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>sessionInfo()</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.5   gdata_2.18.0    dplyr_1.0.8     workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        here_1.0.1        tidyr_1.2.0       getPass_0.2-2    
 [5] xlsxjars_0.6.1    ps_1.6.0          gtools_3.9.2      assertthat_0.2.1 
 [9] rprojroot_2.0.2   digest_0.6.29     utf8_1.2.2        R6_2.5.1         
[13] backports_1.4.1   evaluate_0.15     highr_0.9         httr_1.4.2       
[17] pillar_1.7.0      rlang_1.0.1       rstudioapi_0.13   data.table_1.14.2
[21] car_3.0-12        whisker_0.4       callr_3.7.0       jquerylib_0.1.4  
[25] rmarkdown_2.11    labeling_0.4.2    stringr_1.4.0     munsell_0.5.0    
[29] broom_0.7.12      compiler_4.1.1    httpuv_1.6.5      xfun_0.29        
[33] pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.2  tibble_3.1.6     
[37] fansi_1.0.2       crayon_1.5.0      withr_2.5.0       later_1.3.0      
[41] ggpubr_0.4.0      grid_4.1.1        jsonlite_1.8.0    gtable_0.3.0     
[45] lifecycle_1.0.1   DBI_1.1.2         git2r_0.29.0      magrittr_2.0.2   
[49] scales_1.1.1      xlsx_0.6.5        carData_3.0-5     cli_3.2.0        
[53] stringi_1.7.6     farver_2.1.0      ggsignif_0.6.3    fs_1.5.2         
[57] promises_1.2.0.1  bslib_0.3.1       ellipsis_0.3.2    generics_0.1.2   
[61] vctrs_0.3.8       tools_4.1.1       glue_1.6.2        purrr_0.3.4      
[65] abind_1.4-5       processx_3.5.2    fastmap_1.1.0     yaml_2.3.5       
[69] colorspace_2.0-3  rstatix_0.7.0     rJava_1.0-6       knitr_1.37       
[73] sass_0.4.0       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
